src/snippets/responsive-image.liquid

```diff
@@ -0,0 +1,89 @@
+    - Lazysizes Responsive Images as a Service (https://github.com/aFarkas/lazysizes/tree/gh-pages/plugins/rias) To load the correct image size with our CDN
+    - Lazysizes Bgset (https://github.com/aFarkas/lazysizes/tree/gh-pages/plugins/bgset) To use responsive images on background-image (CSS)
+
+    Accepts:
+    - image: {Object} Image object
+    - width: {Number} Max width of the image container
+    - height: {Number} Max height of the image container
+    - mobile_height: {Number} Max width of the image container in mobile devices
+    - img_class: {String} class name of the <img />
+    - wrapper_class: {String} class name of the <div> wrapper
+    - include_wrapper: {Boolean} Includes <div> wrapper
+
+    Usage:
+    In your liquid template file, copy the following line
+    - {% include 'responsive-image' with image: featured_image, img_class: "css-class", wrapper_class: "wrapper-css-class", width: 700, height: 800, mobile_height: 400, include_wrapper: true %}
+{% endcomment %}
+{% assign mobile_breakpoint = '750px' %}
+
+{% comment %} Added incremental number to avoid conflict styling code when the same images are using this snippet {% endcomment %}
+{% capture responsive_image_counter %}{% increment image_counter %}{% endcapture %}
+
+<style>
+  {% if image.aspect_ratio <= 1 %}
+    {% assign max_height = image.height | at_most: height %}
+    {% assign max_width = max_height | times: image.aspect_ratio %}
+  {% else %}
+    {% assign max_width = image.width | at_most: width %}
+    {% assign max_height = max_width | divided_by: image.aspect_ratio %}
+  {% endif %}
+
+  #ProductImage-{{ image.id }}-{{ responsive_image_counter }} {
+    max-width: {{ max_width }}px;
+    max-height: {{ max_height }}px;
+  }
+  #ProductImageWrapper-{{ image.id }}-{{ responsive_image_counter }} {
+    max-width: {{ max_width }}px;
+  }
+
+  {% comment %}
+    Calculate max height and widths again for small screens
+  {% endcomment %}
+
+  {% if image.aspect_ratio <= 1 %}
+    {% assign max_height = image.height | at_most: mobile_height %}
+    {% assign max_width = max_height | times: image.aspect_ratio %}
+  {% else %}
+    {% assign max_width = image.width | at_most: width %}
+    {% assign max_height = max_width | divided_by: image.aspect_ratio %}
+  {% endif %}
+
+  @media screen and (max-width: {{ mobile_breakpoint}} ) {
+    #ProductImage-{{ image.id }}-{{ responsive_image_counter }} {
+      max-width: {{ max_width }}px;
+      max-height: {{ max_height }}px;
+    }
+    #ProductImageWrapper-{{ image.id }}-{{ responsive_image_counter }} {
+      max-width: {{ max_width }}px;
+    }
+  }
+</style>
+
+{% assign img_url = image | img_url: '1x1' | replace: '_1x1.', '_{width}x.' %}
+
+{% if include_wrapper %}
+  <div id="ProductImageWrapper-{{ image.id }}-{{ responsive_image_counter }}" class="responsive-image__container">
```

Comment:
I don't think we need two wrappers? Do we? Maybe just the one below?

Reply:
Because of the custom `padding-top` calculation, I can't remove one of the two wrappers sadly.